examples/discrete_turbines/channel-optimisation.py

```diff
@@ -0,0 +1,227 @@
+"""
+Basic example of discrete turbine optimisation in a channel
+"""
+from thetis import *
+# this import automatically starts the annotation:
+from firedrake_adjoint import *
+from pyadjoint import minimize
+import numpy as np
+import random
+op2.init(log_level=INFO)
+
+output_dir = 'outputs'
+
+if os.getenv('THETIS_REGRESSION_TEST') is not None:
+    test_gradient = True  # test gradient using Taylor test (see below)
+    optimise = False  # skip actual gradient based optimisation
+else:
+    test_gradient = False
+    optimise = True
+
+# ---- set up the Thetis solver obj as usual ---- #
+if not os.path.exists('mesh.msh'):
+    os.system('gmsh -2  mesh.geo -o mesh.msh')
+mesh2d = Mesh('mesh.msh')
+
+# choose a depth
+H = 50
+h_viscosity = Constant(2.)
+
+# create solver and set options
+solver_obj = solver2d.FlowSolver2d(mesh2d, Constant(H))
+options = solver_obj.options
+# for a steady state solve the timestep is not relevant
+# we just have to ensure that we perform exactly one timestep
+options.timestep = 1.
+options.simulation_export_time = 1.
+options.simulation_end_time = 0.5
+options.output_directory = output_dir
+options.check_volume_conservation_2d = True
+options.element_family = 'dg-cg'
+options.swe_timestepper_type = 'SteadyState'
+# for steady state we use a direct solve (preonly+lu) in combination with a Newton snes solver
+options.swe_timestepper_options.solver_parameters = {'snes_monitor': None,
+                                                     'snes_rtol': 1e-12
+                                                     }
+options.horizontal_viscosity = h_viscosity
+
+options.quadratic_drag_coefficient = Constant(0.0025)
+
+# assign boundary conditions
+left_tag = 1
+right_tag = 2
+coasts_tag = 3
+
+u_inflow = 3.0
+inflow_bc = {'uv': Constant((u_inflow, 0.0))}
+outflow_bc = {'elev': 0.}
+freeslip_bc = {'un': 0.}
+
+solver_obj.bnd_functions['shallow_water'] = {
+    left_tag: inflow_bc,
+    right_tag: outflow_bc,
+    coasts_tag: freeslip_bc
+}
+
+# initialise discrete turbine farm characteristics
```

Comment:
Could you add a bit more explanation in the comments on what these two arrays represent - mention "thrust curve"?